homeassistant/components/signal_messenger/notify.py

```diff
@@ -60,12 +58,12 @@ def send_message(self, message="", **kwargs):
 
         data = kwargs.get(ATTR_DATA)
 
-        filename = None
-        if data is not None and ATTR_FILENAME in data:
-            filename = data[ATTR_FILENAME]
+        filenames = None
```

Comment:
I would prefer that we avoid the breaking change unnecessarily here.

This could be done by allowing both the attribute `filename` as well as the attribute `filenames` to be provided and then internally adding the `filename` to the list of `filenames`.

Reply:
Sounds good to me :+1: I'll adapt the PR, thanks!

Should the `filename` option also be mentioned in the documentation then, or would you see that primarily as a way to stay backwards compatible and as something that's not worth mentioning in the documentation?

Reply:
I would mark it as deprecated and log a warning that it's deprecated and will be removed in version x.y.z (pick a version that's like 3 versions out).

If you are using vol to validate the schema of the service call, we have the deprecated validation helper in `config_validation`

Reply:
Thanks a lot for your suggestions, very much appreciated!

I've updated the PR, please let me know if there's anything else that needs to be changed. 

(I've also removed the `Breaking changes` section from the PR description)